### PR TITLE
BAQE-586: Fix issues revealed by SonarQube

### PR DESCRIPTION
**Pull request by jomarko**
@manstis @hasys would you mind to have a look?

### CODE REVIEW

**Comment by manstis** [comment]
@hasys I did not see where it was changing from "fail" if not present (well, there was potential for Optionals to return `null` and now they're enclosed in `ifPresent()` type code) to ignore. In the cases @jomarko has fixed it is known (by the developer) that the Optionals were present but the safety @jomarko brings is better.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/property/dmn/DefaultValueUtilities.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public static void updateNewNodeName(final Graph<?, Node> graph,
         }
     }
 
-    public static long getMaxUnusedIndex(final Collection<String> values,
+    public static int getMaxUnusedIndex(final Collection<String> values,
```
[the remark this answers is not in the document]
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/property/dmn/DefaultValueUtilities.java]: Thanks, fixed.

**Comment by jomarko** [comment]
@hasys I filed task for me to use your builders once your PR is merged https://issues.jboss.org/browse/BAQE-676

**Comment by jomarko** [comment]
@manstis formatting issue fixed and pushed here.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/property/dmn/DefaultValueUtilities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/property/dmn/DefaultValueUtilities.java]: Is a new commit to follow? The code is still formatted:
```
public static int getMaxUnusedIndex(final Collection<String> values,
                                     final String prefix) {
```

**Comment by jomarko** [comment]
@manstis please merge when you have a moment.

**Pull request merged**